apps/vercel/frontend/src/components/config-screen/ContentTypePreviewPathSection/ContentTypePreviewPathSelectionList/ContentTypePreviewPathSelectionList.tsx

```diff
@@ -0,0 +1,93 @@
+import { ContentType } from '@contentful/app-sdk';
+import { PlusIcon } from '@contentful/f36-icons';
+import { Button } from '@contentful/f36-components';
+import { useState, Dispatch, SetStateAction } from 'react';
+
+import { ContentTypePreviewPathSelectionRow } from '../ContentTypePreviewPathSelectionRow/ContentTypePreviewPathSelectionRow';
+import { ContentTypePreviewPathSelection } from '../../../../types';
+import { actions } from '../../../parameterReducer';
+
+interface Props {
+  contentTypes: ContentType[];
+  dispatchParameters: Dispatch<SetStateAction<any>>;
+  contentTypePreviewPathSelections: ContentTypePreviewPathSelection[];
+}
+
+export const ContentTypePreviewPathSelectionList: React.FC<Props> = ({
+  contentTypes,
+  dispatchParameters,
+  contentTypePreviewPathSelections = [],
+}) => {
+  const [addRow, setAddRow] = useState<number[]>([]);
+
+  const handleUpdateParameters = (parameters: ContentTypePreviewPathSelection) => {
+    if (parameters.contentType && parameters.previewPath) {
+      if (addRow.length) setAddRow([]);
+      dispatchParameters({
+        type: actions.ADD_CONTENT_TYPE_PREVIEW_PATH_SELECTION,
+        payload: parameters,
+      });
+    }
+  };
+
+  const handleRemoveRow = (parameters: ContentTypePreviewPathSelection) => {
+    if (addRow.length) setAddRow([]);
+
+    dispatchParameters({
+      type: actions.REMOVE_CONTENT_TYPE_PREVIEW_PATH_SELECTION,
+      payload: parameters,
+    });
+  };
+
+  // TO DO: Adjust AddRow logic to boolean if we want to enforce only one row appearing at once
+  const handleAddRow = () => {
+    setAddRow([...addRow, addRow.length]);
+  };
+
+  const isAddButtonDisabled =
+    Boolean(addRow.length) || !Boolean(contentTypePreviewPathSelections.length);
+
+  const renderSelectionRow = () => {
+    // TO DO: Handle case where contentTypes are not present - do not render add button etc.
```

Comment:
**design follow-up:**  for how to handle no content types. We render something within the dropdown right now when there are not content types. Does that suffice? I feel the 'Add Content Type' button might be confusing in these cases.